File: frontend/app/src/api/members.ts
import { apiGet } from './client';
import type { components } from '../shared/contracts/api';

export type MemberProfileResponse = components['schemas']['MemberProfileResponse'];
export type MemberSummary = components['schemas']['MemberSummary'];
export type MemberActivityResponse = components['schemas']['MemberActivityResponse'];
export type MemberActivityItem = components['schemas']['MemberActivityItem'];
export type MembersResponse = {
    members: MemberSummary[];
    total: number;
    page: number;
    limit: number;
};

export type MemberFilters = {
    query?: string;
    state?: string;
    chamber?: string;
    party?: string;
};

export function fetchMembers(page = 1, limit = 50, filters: MemberFilters = {}): Promise<MembersResponse> {
    const params = new URLSearchParams({ page: String(page), limit: String(limit) });
    if (filters.query?.trim()) params.set('query', filters.query.trim());
    if (filters.state) params.set('state', filters.state);
    if (filters.chamber) params.set('chamber', filters.chamber);
    if (filters.party) params.set('party', filters.party);
    return apiGet<MembersResponse>(`/api/v1/members?${params.toString()}`);
}

export function fetchMemberProfile(bioguideId: string): Promise<MemberProfileResponse> {
    return apiGet<MemberProfileResponse>(`/api/v1/members/${bioguideId}`);
}

export function fetchMemberActivity(
    bioguideId: string,
    page = 1,
    limit = 25,
    types?: string,
): Promise<MemberActivityResponse> {
    const params = new URLSearchParams({ page: String(page), limit: String(limit) });
    if (types) params.set('types', types);
    return apiGet<MemberActivityResponse>(
        `/api/v1/members/${bioguideId}/activity?${params.toString()}`,
    );
}

File: cdm/backend/api/routes/members.py
from fastapi import APIRouter, HTTPException, Query

from cdm.backend.services.member_service import (
    get_member_profile,
    list_member_activity,
    list_members,
)
from cdm.backend.services.topic_service import get_member_topics
from cdm.contracts.api import (
    MemberActivityResponse,
    MemberProfileResponse,
    MembersResponse,
    MemberTopicsResponse,
)

router = APIRouter(prefix="/members", tags=["members"])


@router.get("", response_model=MembersResponse)
def get_members(
    query: str | None = Query(default=None),
    state: str | None = Query(default=None),
    chamber: str | None = Query(default=None),
    party: str | None = Query(default=None),
    page: int = Query(default=1, ge=1),
    limit: int = Query(default=50, ge=10, le=100),
) -> MembersResponse:
    return list_members(query, state, chamber, party, page, limit)


@router.get("/{bioguide_id}", response_model=MemberProfileResponse)
def get_member(bioguide_id: str) -> MemberProfileResponse:
    try:
        return get_member_profile(bioguide_id)
    except ValueError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc


@router.get("/{bioguide_id}/activity", response_model=MemberActivityResponse)
def get_member_activity(
    bioguide_id: str,
    types: str | None = Query(default=None, description="Comma-separated document types"),
    page: int = Query(default=1, ge=1),
    limit: int = Query(default=25, ge=5, le=100),
) -> MemberActivityResponse:
    return list_member_activity(bioguide_id, types, page, limit)


@router.get("/{bioguide_id}/topics", response_model=MemberTopicsResponse)
def member_topics(bioguide_id: str) -> MemberTopicsResponse:
    return get_member_topics(bioguide_id)
